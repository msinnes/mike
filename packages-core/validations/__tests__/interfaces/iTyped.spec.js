const { isInterface, loadType } = require('@mike/class');

const iTyped = require('../../interfaces/iTyped');

describe('iTyped', () => {
  it('should be an interface', () => {
    expect(isInterface(iTyped)).toBe(true);
  });

  it('should accept a valid config', () => {
    expect(() => {
      iTyped.ensure({ type: loadType('name', () => {})});
    }).not.toThrow();
  });

  it('should throw an error if the config is invalid', () => {
    expect(() => {
      iTyped.ensure({});
    }).toThrowErrorMatchingSnapshot();
  });
});
