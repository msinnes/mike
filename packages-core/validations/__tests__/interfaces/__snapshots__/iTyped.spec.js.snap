// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iTyped should throw an error if the config is invalid 1`] = `
"ClassError: Runtime Interface Error
type: Expected Type"
`;
